### PR TITLE
[Movement v1] SfP and Live Viewer shortcuts

### PR DESCRIPTION
**Pull request by maltewae**
This PR adds two optional shortcuts in the Coordinate Pairing Window for Search for Peak and Live Viewer. This makes pairing creation easier.

Screenshot 2022-11-30 at 14 33 46

In the future, when we define the new interface for SfP, we can directly set the required stages to perform the SfP Algorithm.

### CODE REVIEW

File: LabExT/View/Movement/CoordinatePairingsWindow.py [hunk 1]
```diff
@@ -161,6 +172,24 @@ def __setup__(self) -> None:
...
                 side=TOP, fill=X)
 
+        if self.experiment_manager:
+            shortcuts_frame = CustomFrame(self._main_frame)
+            shortcuts_frame.pack(side=TOP, fill=X, pady=5)
+
+            search_for_peak_button = Button(
+                shortcuts_frame,
+                text="Perform Search for Peak...",
+                command=self._on_perform_search_for_peak)
```
Comment (marcoep) [inline LabExT/View/Movement/CoordinatePairingsWindow.py:182]: Suggestion: I would use the function `self.experiment_manager.main_window.open_live_viewer` here, so you can re-use all the top-level checking logic from the main window.

File: LabExT/View/Movement/CoordinatePairingsWindow.py [hunk 2]
```diff
@@ -161,6 +172,24 @@ def __setup__(self) -> None:
...
+            search_for_peak_button.pack(
+                side=RIGHT, fill=Y, pady=5, expand=0)
+
+            live_viewer_button = Button(
+                shortcuts_frame,
+                text="Open Live Viewer...",
+                command=self._on_open_live_viewer)
```
Comment (marcoep) [inline LabExT/View/Movement/CoordinatePairingsWindow.py:189]: Same here, reuse the function `self.experiment_manager.main_window.open_peak_searcher` with no need to re-code logic in this class.

File: LabExT/View/Movement/CoordinatePairingsWindow.py [hunk 3]
```diff
@@ -252,7 +281,36 @@ def _ask_user_move_to_device(self):
...
+        if not self.experiment_manager:
+            return
+
+        if try_to_lift_window(self._search_for_peak_toplevel):
+            return
+
+        sfpp = SearchForPeakPlotsWindow(
+            parent=self,
+            experiment_manager=self.experiment_manager)
+        self._search_for_peak_toplevel = sfpp.plot_window
```
Comment (marcoep) [inline LabExT/View/Movement/CoordinatePairingsWindow.py:313]: same logic as in `LabExT/View/MenuListener.py`, no need to re-code

**Review by marcoep — CHANGES_REQUESTED** [review]

**Comment by maltewae** [comment]
Neat! Didn't saw the methods on main window level. Changed it 😄

File: LabExT/View/Movement/CoordinatePairingsWindow.py [hunk 4]
```diff
@@ -161,6 +166,24 @@ def __setup__(self) -> None:
...
                 side=TOP, fill=X)
 
+        if self.experiment_manager:
+            shortcuts_frame = CustomFrame(self._main_frame)
+            shortcuts_frame.pack(side=TOP, fill=X, pady=5)
+
+            search_for_peak_button = Button(
+                shortcuts_frame,
+                text="Perform Search for Peak...",
+                command=self.experiment_manager.main_window.open_peak_searcher())
```
Comment (marcoep) [inline LabExT/View/Movement/CoordinatePairingsWindow.py:176]: I think you only need to pass the function, not the function-call's-return here. i.e. delete parentheses at end.

File: LabExT/View/Movement/CoordinatePairingsWindow.py [hunk 5]
```diff
@@ -161,6 +166,24 @@ def __setup__(self) -> None:
...
+            search_for_peak_button.pack(
+                side=RIGHT, fill=Y, pady=5, expand=0)
+
+            live_viewer_button = Button(
+                shortcuts_frame,
+                text="Open Live Viewer...",
+                command=self.experiment_manager.main_window.open_live_viewer())
```
Comment (marcoep) [inline LabExT/View/Movement/CoordinatePairingsWindow.py:183]: I think you only need to pass the function, not the function-call's-return here. i.e. delete parentheses at end.

**Review by marcoep — CHANGES_REQUESTED** [review]

File: LabExT/View/Movement/CoordinatePairingsWindow.py (hunk 5 above)
Comment (maltewae) [reply LabExT/View/Movement/CoordinatePairingsWindow.py]: Yes! 😅 I was too fast.

**Review by marcoep — APPROVED** [review]

**Pull request merged**